examples/app/stock_applet/stock_app.py

```diff
@@ -198,10 +217,13 @@ def input_change(self, obj, attrname, old, new):
             self.ticker2 = new
         if obj == self.ticker1_select:
             self.ticker1 = new
-        self.make_source()
-        self.make_plots()
-        self.set_children()
-        curdoc().add(self)
+
+        # avoid trying to compare same stocks
+        if self.ticker2 != self.ticker1:
```

Comment:
Yeah, I started to write what I was calling an ExclusiveSelectGroup, but
not really sure where that would fit. I think this happens a lot where you
may say it is only valid to compare things that are different. So you need
to manage the state of the available selections for all selectors. Doing
this manually is kind of messy and it seems like it would often be needed.

As an interim, I at least made it not error out. Is there a popup dialog
widget, like what bootstrap calls a modal?:
http://getbootstrap.com/javascript/#modals This would be better than trying
to output some message in the GUI somewhere.

On Wed, Jan 28, 2015 at 1:39 PM, Damián Avila <notifications@github.com>
wrote:

> In examples/app/stock_applet/stock_app.py
> <https://github.com/bokeh/bokeh/pull/1764#discussion_r23715983>:
>
> > @@ -198,10 +217,13 @@ def input_change(self, obj, attrname, old, new):
> >              self.ticker2 = new
> >          if obj == self.ticker1_select:
> >              self.ticker1 = new
> > -        self.make_source()
> > -        self.make_plots()
> > -        self.set_children()
> > -        curdoc().add(self)
> > +
> > +        # avoid trying to compare same stocks
> > +        if self.ticker2 != self.ticker1:
>
> Not sure about this... when I try it and I did not see any update or any
> message about I am trying to compare the same stock... seems that the app
> is not working... I guess I prefer the comparative between the same stock,
> even if it is redundant... other @bokeh/core
> <https://github.com/orgs/bokeh/teams/core> thoughs?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bokeh/bokeh/pull/1764/files#r23715983>.
>